### DISCUSSION TITLE
should we make a dev.jbang.ext jar library that provides some basic 'os' / terminal functions?

### DISCUSSION THREAD

**Discussion by maxandersen** [jbangdev/jbang#978] [Ideas]
https://github.com/google/zx which is a 'use node.js to write scripts` kinda thing made me play with https://gitlab.com/ongresinc/fluent-process to which I added a `$()` method to allow simpler shell calls in java:

```
$("echo hello; echo world")
                        .pipe$("cat")
                        .stream()
```

perfect for jbang but its in custom fork atm.

in zx they also have easy method for fetching http requests, ie. `fetch('http://wttr.in')` and question methods which https://github.com/awegmann/consoleui maybe be a good alternative for, minimist for parsing...picocli, and a few other bits

we could recommend user to import all these various packages but i'm wondering if we should make a maven module that import them all at once and/or even fork or wrap them to make a simpler unified version of them available under `dev.jbang.ext` or similar ? 

....what I was thinking was if we had this set of "good" libs then make it so if you named your .jsh file for .jshx then we would automatically add them and thus make the files even shorter/simpler.

For now just an idea that been bumping around my head.....wdyt ?

**Comment by quintesse** [comment]
YEs, that has always seemed like an interesting idea to me.

**Reply by maxandersen** [reply]
So fork, wrap or depend ?

**Reply by quintesse** [reply]
Personally I'd say depend first and wrap where it results in much simpler code. Forking seems a bit of a maintenance nightmare to me.

**Reply by maxandersen** [reply]
yeah but i.e. fluentprocess is implemented quite protective so need to be a fork if wanna get the shell functionality.

**Reply by quintesse** [reply]
I what way is it protective?

**Reply by maxandersen** [reply]
its classes a package protected.

**Comment by maxandersen** [comment]
also just realized java does not have a decent way to show an image.

python its: image.show();

java is 10ish+ lines of jframe/awt mess.

using/recommending imagej might be the way forward there.
